src/wfc/collapse.rs

```diff
@@ -189,7 +194,7 @@ pub fn collapse(
     mut output_graph: Graph,
     seed: Option<u64>,
     iterations: Option<usize>
-) -> Graph {
+) -> Vec<Graph> {
```

Comment:
I think this function should just return the vertices, as it is currently we do a lot of redundant `clones` which could possibly take quite a while depending on the size of the graph. More below...